distil/primitives/satellite_image_loader.py

```diff
@@ -9,15 +10,50 @@
 from common_primitives import base
 from d3m import container, utils as utils
 from d3m.base import utils as base_utils, primitives
-from d3m.metadata import base as metadata_base
+from d3m.metadata import base as metadata_base, hyperparams
 from d3m.primitive_interfaces import base as base_prim
+from d3m.primitive_interfaces import transformer
 from distil.primitives import utils as distil_utils
 from distil.utils import CYTHON_DEP
 import version
+import lzo
+import struct
 
 logger = logging.getLogger(__name__)
 
-class DataFrameSatelliteImageLoaderPrimitive(primitives.FileReaderPrimitiveBase):
+class Hyperparams(hyperparams.Hyperparams):
+    use_columns = hyperparams.Set(
+        elements=hyperparams.Hyperparameter[int](-1),
+        default=(),
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/ControlParameter'],
+        description="A set of column indices to force primitive to operate on. If any specified column does not contain filenames for supported media types, it is skipped.",
+    )
+    exclude_columns = hyperparams.Set(
+        elements=hyperparams.Hyperparameter[int](-1),
+        default=(),
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/ControlParameter'],
+        description="A set of column indices to not operate on. Applicable only if \"use_columns\" is not provided.",
+    )
+    return_result = hyperparams.Enumeration(
+        values=['append', 'replace', 'new'],
+        default='append',
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/ControlParameter'],
+        description="Should columns with read files be appended, should they replace original columns, or should only columns with read files be returned?",
+    )
+    add_index_columns = hyperparams.UniformBool(
+        default=True,
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/ControlParameter'],
+        description="Also include primary index columns if input data has them. Applicable only if \"return_result\" is set to \"new\".",
+    )
+    compress_data = hyperparams.Hyperparameter[bool](
+        default=False,
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/ControlParameter'],
+        description="If True, applies LZO compression algorithm to the data"
```

Comment:
A description of the byte format of the compressed data should be included here.  We've got it as a comment where it does the compression, but it should be here at the parameter level since its part of the public interface of the primitive.